engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/sink/AvroSinkValueParameter.scala

```diff
@@ -0,0 +1,83 @@
+package pl.touk.nussknacker.engine.avro.sink
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CustomNodeError, NodeId}
+import pl.touk.nussknacker.engine.api.definition.Parameter
+import pl.touk.nussknacker.engine.api.typed
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedClass, TypedObjectTypingResult, TypedUnion, TypingResult}
+import pl.touk.nussknacker.engine.avro.KafkaAvroBaseTransformer.{SchemaVersionParamName, SinkKeyParamName, SinkValidationModeParameterName, SinkValueParamName, TopicParamName}
+import pl.touk.nussknacker.engine.avro.sink.AvroSinkValueParameter.FieldName
+import pl.touk.nussknacker.engine.definition.parameter.editor.ParameterTypeEditorDeterminer
+
+private[sink] case object AvroSinkValueParameter {
+
+  type FieldName = String
+
+  private val restrictedParamNames = Set(SchemaVersionParamName, SinkKeyParamName, SinkValidationModeParameterName, TopicParamName)
+
+  def apply(typing: TypingResult)(implicit nodeId: NodeId): Validated[ProcessCompilationError, AvroSinkValueParameter] =
+    toSinkValueParameter(typing, paramName = None, isTopLevel = true)
+
+  private def toSinkValueParameter(typing: TypingResult, paramName: Option[String], isTopLevel: Boolean)
+                                  (implicit nodeId: NodeId): Validated[ProcessCompilationError, AvroSinkValueParameter] =
+    typing match {
+      case typed.typing.Unknown =>
+        Invalid(CustomNodeError(nodeId.id, "Cannot determine typing for provided schema", None))
+
+      case typedObject: TypedObjectTypingResult if containsRestrictedNames(typedObject) =>
```

Comment:
Can we add TODO to handle this case somehow?

Reply:
:+1: 
https://github.com/TouK/nussknacker/pull/1405/commits/c9c38c070bd37ae7273f0f486f4a95700033e7bc